src/install/bin.zig

```diff
@@ -352,6 +355,193 @@ pub const Bin = extern struct {
+            \\  SET "_prog=node"
+            \\  SET PATHEXT=%PATHEXT:;.JS;=;%
+            \\)
+            \\
+            \\endLocal & goto #_undefined_# 2>NUL || title %COMSPEC% & "%_prog%"  "%dp0%\
+        ;
+
+        const cmd_contents_end =
+            \\" %*
+        ;
+
+        var cmd_contents_buf = brk: {
+            // path to file is only used once. copy into a stack allocated buffer
+            var buf: [cmd_contents_begin.len + cmd_contents_end.len + bun.MAX_PATH_BYTES]u8 = undefined;
+            @memcpy(buf[0..cmd_contents_begin.len], cmd_contents_begin);
+            break :brk buf;
+        };
+
+        fn createCmdFile(this: *Linker, target_path: [:0]const u8, dest_path_buf: *bun.PathBuffer, dest_path: [:0]const u8) void {
+            const ext = this.cmd_ext orelse brk: {
+                if (bun.getenvZ("PATHEXT")) |pathext| {
+                    var iter = std.mem.splitScalar(u8, pathext, std.fs.path.delimiter);
+                    while (iter.next()) |ext| {
+                        // https://github.com/pnpm/pnpm/issues/3800
+                        // https://github.com/zkochan/packages/blob/0397b5df8b4233a644ea45639ddb1c06f93b24c1/cmd-extension/index.js#L9
+                        if (strings.eqlCaseInsensitiveASCII(ext, ".cmd", true)) {
+                            this.cmd_ext = ext;
+                            break :brk this.cmd_ext.?;
+                        }
+                    }
+                }
+
+                this.cmd_ext = ".cmd";
+                break :brk this.cmd_ext.?;
+            };
+
+            @memcpy(dest_path_buf[dest_path.len..][0..ext.len], ext);
+            const dest = dest_path_buf[0 .. dest_path.len + ext.len];
+
+            const root_dir = std.fs.Dir{ .fd = bun.fdcast(this.package_installed_node_modules) };
+            const cmd_file = root_dir.createFile(dest, .{}) catch |err| {
+                this.err = err;
+                return;
+            };
+
+            const cmd_file_fd = bun.toLibUVOwnedFD(cmd_file.handle);
+            defer _ = bun.sys.close(cmd_file_fd);
+
+            @memcpy(cmd_contents_buf[cmd_contents_begin.len..][0..target_path.len], target_path);
+            @memcpy(cmd_contents_buf[cmd_contents_begin.len + target_path.len ..][0..cmd_contents_end.len], cmd_contents_end);
+
+            const contents = cmd_contents_buf[0 .. cmd_contents_begin.len + target_path.len + cmd_contents_end.len];
+
+            var index: usize = 0;
+            while (index < contents.len) {
+                switch (bun.sys.write(bun.toFD(cmd_file_fd), contents[index..])) {
+                    .result => |written| {
+                        index += written;
+                    },
+                    .err => |err| {
+                        this.err = @errorFromInt(err.errno);
+                        return;
+                    },
+                }
+            }
+        }
+
+        const ps1_contents_fmt =
+            \\#!/usr/bin/env pwsh
```

Comment:
@zackradisic wdyt about using bun shell here

Reply:
Would need to support if statements (e.g. `if [ -x "$basedir/node" ]; then...`)

Would need $0 so `basedir=$(dirname echo $0)` works

And also would be nice to have `dirname` implemented in bun shell. Maybe we can have a custom command for getting the basedir itself